servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancerBuilder.java

```diff
@@ -148,6 +160,126 @@ public LoadBalancerFactory<ResolvedAddress, C> build() {
                 loadBalancerObserver, healthCheckerSupplier);
     }
 
+    public static final class DefaultHttpLoadBalancerFactory<ResolvedAddress> implements HttpLoadBalancerFactory<ResolvedAddress> {
```

Comment:
In my minds eye, making a new `DefaultHttpLoadBalancerFactory` makes it tough for users to switch to the new load balancer. The `HttpClients` class uses the `DefaultHttpLoadBalancerFactory` internally: could we add the piping to that class? I imagine we would need to massage both `toLoadBalancedConnection` methods at that point but it seems fine to do. I think we'd need to open up a few more types to being public.